### ISSUE TITLE
Should CodeMeta include the concept of "mailing list" / "discussion forum" in next release

### ISSUE THREAD

**Issue by npch** [codemeta/codemeta#166]
As noted by @moranegg and @sdruskat at the FORCE2017 Hackathon, many pieces of software will have a forum for discussion, typically a mailing list or web-based discussion forum or online chat. Should this be represented in the codemeta as an intrinsic part of the codemeta:SoftwareSourceCode?

**Comment by cboettig** [comment]
Sounds like a good idea.  (we have `issueTracker`, but obviously many projects have both).  Not seeing an obvious schema.org term for this concept; any suggestions?  IIRC, I think there is a mailing list concept in doap ontology...

**Comment by npch** [comment]
The [DOAP schema](http://usefulinc.com/ns/doap) supports both "mailing-list" and "support-forum" (as well as "developer-forum".

In the schema.org SubscribeAction definition example of subscribing to a mailing list, the mailing list is identified as type "Product".

**Comment by cboettig** [comment]
Right, so the schema.org doesn't really give us a term that could be considered a property of a SoftwareSourceCode or SoftwareApplication type object.  There's also the notion of https://schema.org/contactPoint, which is a property of Person or Organization, which is clearly a more commercial flavor of the concept; but compare Google's recommended use of contactPoint in Dataset: https://developers.google.com/search/docs/data-types/datasets)


My intuition is that the schema-y thing to do is to have a generic as possible property for this, (possibly with optional refinements).  As a general rule of thumb I think being able to aggregate related concepts is often more relevant to use cases than being able to disambiguate -- knowing the difference between the "developer mailing list" and the "user mailing list" is important when you're considering an individual project that has both; but then you can usually figure that out from the more natural documentation from that particular project and not the metadata.  When trying to do tooling that consumes a wide variety of codemeta.json files and say, points people to some point of contact, too many snowflake cases just defeat the purpose of a common schema to begin with.  

Perhaps the right way forward would be to generalize our notion of `codemeta:issueTracker` then?  I'm not sure.

**Comment by npch** [comment]
I think this is one case where I think we just need to try this out in implementations and see what does or doesn't work? 

I agree that we don't want to proliferate many different mailing lists, so perhaps we're looking to try examples from real life and see what concepts they need and how these are represented when things are being exchanged from system to system.

**Comment by cboettig** [comment]
👍

**Comment by moranegg** [comment]
This issue will be open for discussion until March 15th and we will start a vote from March 15th until March 25th 2023

Since there are many projects using mailing lists and issue-trackers, my vote is to add a mailing-list property.

**Comment by proycon** [comment]
I agree with what @cboettig wrote back in 2017 about generalizing the notion of
`codemeta:issueTracker`, but doing that retroactively now is properly no longer
a good idea. An issue tracker is a pretty specific thing and has been in use
for a while already now. Because it's fairly specific it means we're missing
properties for mailing-lists, chat channels, or discussion forums currently.

I think a key point to decide on is whether we want to distinguish by interface
type/modality of interaction (mail, web-based ticketing system, chat channel)
or by the semantics of what a particular support channel is used for (user
support, user discussion, developer discussion & patch submission).
The current `issueTracker` kind of fits both so we could still go either way.

Adding a `mailingList` property, as suggested here, would go the first route and leaves in the
middle what the list is used for exactly (which is fine), but that means we do need extra
properties like `webForum` and `chatChannel` (just inventing names here) for things that are not
mailing lists and provide another interface.

If we go the second route we'd have to think about the kinds of different
support modalities that exist, but this may be more complicated as often the
lines are blurred. So I'd vote for the first option, but suggest we don't just
do just mailing lists but make sure web forums (e.g discourse/github discussion) and support chat channels
(e.g. irc/matrix/gitter/telegram/slack/discord/etc etc) have their own property
too then.

**Comment by dgarijo** [comment]
ContactPoint seems to work. By extending the domain of `contactPoint` we can do interesting things, even adding more than one contact point (supporting both mailing lists and discord channels):
```
{ "@context" : "https://schema.org",
  "@id": mySoftware,
  "@type" : "SoftwareSourceCoce",
  "contactPoint" : [
    { "@type" : "ContactPoint",
        "url" : "http://discordURL.com",
      "contactType" : "Discord channel support"
    },
   { "@type" : "ContactPoint",
        "email" : "list@mailinglist.com",
      "contactType" : "Mailing list support"
    }  ] }
```
(I haven't validated the json-ld, it may contain typos)
